### ISSUE TITLE
rusty-demo fails to build on windows

### ISSUE THREAD

**Issue by MolotovCherry** [hermitcore/rusty-hermit#413]
I have the required things installed.
- nasm (in PATH as well)
- rust toolchain `nightly-2023-03-15` installed and ready to go (with `rust-src` component)
- rusty-demo lists (as of this current date):
```toml
[target.'cfg(target_os = "hermit")'.dependencies]
hermit-sys = "0.4"
```

The build fails with
```
error[E0635]: unknown feature `atomic_mut_ptr`
    --> src\lib.rs:12:12
     |
  12 | #![feature(atomic_mut_ptr)]
     |            ^^^^^^^^^^^^^^

```

Changing `rust-demo` `hermit-sys` to the following fixes it.
`hermit-sys = { git = "https://github.com/hermitcore/rusty-hermit" }`

Don't know where, but something must be going wrong somewhere, but it's fixed in master. Theoretically I don't think this is a problem with the demo itself. if you use the crates.io package at all, this happens.

**Comment by stlankes** [comment]
Oh, we have to update `hermit-sys` at crated.io. I will do it tomorrow.

**Comment by mkroening** [comment]
Ah, I understand what's going wrong. Apparently, windows does not respect the internal `rust-toolchain.toml` in `libhermit-rs`.

You should be able to work around this issue by downgrading to `nightly-2022-10-19`, while we figure out a proper fix. 🤔

**Comment by mkroening** [comment]
Ah, yes. This sounded familiar. Manually installing the `llvm-tools-preview` component for that toolchain should work around the issue. (https://github.com/hermitcore/rusty-hermit/issues/374#issuecomment-1289216468)

I don't have a local windows setup, but I'll continue experimenting on what we have to change in the `build.rs` of `hermit-sys`, so automatic `rust-toolchain.toml` detection for the kernel works on windows as well: https://github.com/hermitcore/rusty-demo/pull/26

**Comment by MolotovCherry** [comment]
> llvm-tools-preview

Confirmed that `llvm-tools-preview` fixes the issue!

**Comment by mkroening** [comment]
I fixed the underlying issue in https://github.com/hermitcore/rusty-hermit/pull/431. 🎉

**Comment by mkroening** [comment]
Published as hermit-sys 0.5.1. :)
